From: "Barak A. Pearlmutter" <barak+git@cs.nuim.ie>
Date: Tue, 5 Jun 2012 12:41:08 +0100
Subject: icu-config ldflags

Switch invocation icu-config --ldflags option to three options, to
work around recent change in icu-config that includes options used
only in creating dynamic libraries.  This patch can be removed if
icu-config --ldflags goes back to only outputting user-level options.
---
 config.pkg |    9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/config.pkg b/config.pkg
index 6d42c35..db49a75 100644
--- a/config.pkg
+++ b/config.pkg
@@ -193,6 +193,7 @@ EOF
 
 	if [ "yes" = "${with_unicode}" ] ; then
 		ICU_CONFIG="${option_icu_config:-icu-config}"
+		ICU_CONFIG_LDFLAGS="--ldflags-searchpath --ldflags-libsonly --ldflags-system"
 		{
 			{
 				check_message "Checking for icu-config... "
@@ -203,8 +204,8 @@ EOF
 					exit 1
 				fi
 
-				echo "Running ${ICU_CONFIG} --cppflags --ldflags" >> config.log
-				if "${ICU_CONFIG}" --cppflags --ldflags 2>&1 > /dev/null ; then
+				echo "Running ${ICU_CONFIG} --cppflags ${ICU_CONFIG_LDFLAGS}" >> config.log
+				if "${ICU_CONFIG}" --cppflags ${ICU_CONFIG_LDFLAGS} 2>&1 > /dev/null ; then
 					check_message_result "yes"
 					true
 				else
@@ -244,7 +245,7 @@ int main(int argc, char *argv[]) {
 }
 EOF
 				clean .config.o
-				test_link "functions in libicu" ICUFLAGS="\`\"${ICU_CONFIG}\" --cppflags\`" ICULIBS="\`\"${ICU_CONFIG}\" --ldflags\`"
+				test_link "functions in libicu" ICUFLAGS="\`\"${ICU_CONFIG}\" --cppflags\`" ICULIBS="\`\"${ICU_CONFIG}\" ${ICU_CONFIG_LDFLAGS}\`"
 			}
 		} || {
 			check_message_result "!! Could not compile with Unicode support. Try configuring with --without-unicode."
@@ -252,7 +253,7 @@ EOF
 		}
 
 		ICUFLAGS="-DUSE_UNICODE \`\"${ICU_CONFIG}\" --cppflags\`"
-		ICULIBS="\`\"${ICU_CONFIG}\" --ldflags\`"
+		ICULIBS="\`\"${ICU_CONFIG}\" ${ICU_CONFIG_LDFLAGS}\`"
 		if checkfunction "nl_langinfo" "nl_langinfo(CODESET);" "<langinfo.h>" ; then
 			ICUFLAGS="${ICUFLAGS} -DUSE_NL_LANGINFO"
 		fi
